frontend/src/components/AgentControlBar.tsx

```diff
@@ -72,13 +76,24 @@ function ActionButton({
 
 ActionButton.defaultProps = {
   large: false,
+  isLoading: false,
 };
 
 function AgentControlBar() {
+  const { initialized } = useSelector((state: RootState) => state.task);
   const { curTaskState } = useSelector((state: RootState) => state.agent);
   const [desiredState, setDesiredState] = React.useState(AgentTaskState.INIT);
   const [isLoading, setIsLoading] = React.useState(false);
 
+  const handleReset = () => {
+    if (!initialized) return;
+
+    initializeAgent(getSettings());
+    // act = AgentTaskAction.STOP;
+    clearMsgs().then().catch();
```

Comment:
Prefer `async...await`. It is a cleaner implementation over an empty `then...catch`

Reply:
let me adjust it